compiler/src/emit.rs

```diff
@@ -68,151 +82,135 @@ fn emit_literal(literal: &LiteralValue, instructions: &mut Instructions, cp: &mu
     };
 }
 
-fn emit_ref(
-    symbol: &Symbol,
-    ref_type: TypeId,
-    instructions: &mut Instructions,
-    locals: &LocalsLayout,
-) {
-    match symbol {
-        Symbol::Local(id) => {
-            let size = ref_type.into();
-            instructions.emit_get_local(*id, size, locals);
-        }
-        _ => todo!(),
-    }
-}
-
 #[allow(clippy::too_many_arguments)]
 fn emit_declaration(
     declaration: &Declaration,
     instructions: &mut Instructions,
-    typing: &Typing,
     engine: &Engine,
     cp: &mut ConstantPool,
     locals: &mut LocalsLayout,
     state: &mut EmissionState,
+    captures: &mut Captures,
```

Comment:
```suggestion
    captures: &Captures,
```
In your current implementation, you don't need a mutable reference. This is however due to the computations done ahead of time that shouldn't be possible.